compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -119,13 +118,9 @@ unsafe fn configure_llvm(sess: &Session) {
     llvm::LLVMInitializePasses();
 
     for plugin in &sess.opts.debugging_opts.llvm_plugins {
-        let path = Path::new(plugin);
-        let res = DynamicLibrary::open(path);
-        match res {
-            Ok(_) => debug!("LLVM plugin loaded succesfully {} ({})", path.display(), plugin),
-            Err(e) => bug!("couldn't load plugin: {}", e),
-        }
-        mem::forget(res);
+        let lib = Library::new(plugin).unwrap_or_else(|e| bug!("couldn't load plugin: {}", e));
+        debug!("LLVM plugin loaded successfully {:?} ({})", lib, plugin);
+        mem::forget(lib);
```

Comment:
Could you add a comment on how this loop works? Does it rely on some load-time execution inside the library?

Reply:
Yes, LLVM plugins use global value initializers to register themself with LLVM. Something like

```cpp
static RegisterPass<Hello> X("hello", "Hello World Pass",
                             false /* Only looks at CFG */,
                             false /* Analysis Pass */);
```

I believe: https://www.llvm.org/docs/WritingAnLLVMPass.html#basic-code-required

Reply:
Added a brief comment.